qiita_db/meta_util.py

```diff
@@ -163,3 +168,34 @@ def get_accessible_filepath_ids(user):
             filepath_ids.update(analysis.all_associated_filepath_ids)
 
         return filepath_ids
+
+
+def get_lat_longs():
+    """Retrieve the latitude and longitude of all the samples in the DB
+
+    Returns
+    -------
+    list of [float, float]
+        The latitude and longitude for each sample in the database
+    """
+    portal_table_ids = Portal(qiita_config.portal).get_studies()
+
+    with TRN:
+        sql = """SELECT DISTINCT table_name
+                 FROM information_schema.columns
+                 WHERE SUBSTR(table_name, 1, 7) = 'sample_'
+                    AND table_schema = 'qiita'
+                    AND column_name IN ('latitude', 'longitude');"""
+        TRN.add(sql)
+
+        sql = "SELECT latitude, longitude FROM qiita.{0}"
+        idx = TRN.index
+
+        portal_tables = TRN.execute_fetchflatten()
+        portal_tables = [x for x in TRN.execute_fetchflatten()
```

Comment:
I made the change -- you may be right, since terrance/phillip database server is probably more efficient than the VM on which the the python code would be running. Not sure, but it's an easy enough change to make.